### PR TITLE
Milestone 07: Unit tests with Rspec

### PR DESCRIPTION
**Pull request by jmoseka**
## Unit tests with Rspec

**This branch includes:**

- [x] Add unit tests to your classes:
   - `Book`
   - `Person`
   - `Student`
   -  `Teacher`
   -  `Classroom`
   -  `Rental`
   -  all classes created are implementing the Decorator design pattern

### CODE REVIEW

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @jmoseka and @mavericks-db, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Nice collaborative work :100: 
- [ ] All methods in the classes are tested :fireworks: 
- [ ] All test files are in the spec directory :1st_place_medal: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/rental_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+require '../book'
+require '../teacher'
+require '../rental'
+
```
[the remark this answers is not in the document]
Comment (mavericks-db) [reply spec/rental_spec.rb:4]: Done in commit https://github.com/mavericks-db/oop-school-library/pull/7/commits/2941a955b13c20f5f353908d3e56bfbe4f5d065c

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+require '../book'
```
[the remark this answers is not in the document]
Comment (mavericks-db) [reply spec/book_spec.rb]: Changes applied to all _spec.rb files

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+require '../book'
+
+describe 'Person object' do
+  context 'should create an instance of Person class' do
+    person = Person.new(15, 'John', 'true')
+
+    it 'returns a Person object' do
+      expect(person.class) == Person
+    end
+
```
[the remark this answers is not in the document]
Comment (mavericks-db) [reply spec/person_spec.rb:12]: Fix on commit https://github.com/mavericks-db/oop-school-library/pull/7/commits/cb2d18f9e29a552ca33514f49be5520890278536

**Review by ShadyShawkat — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mavericks-db** [comment]
Thank you 👍

**Pull request merged**